obs2ioda-v2/src/f_c_string_t_mod.f90

```diff
@@ -78,12 +78,12 @@ function to_c(this, f_string) result(c_string)
         character(len = *), intent(in) :: f_string
         type(c_ptr) :: c_string
         integer :: n
-        n = len(f_string)
+        n = len(trim(f_string))
```

Comment:
Although the code is correct as written, you could consider using the `len_trim` intrinsic function.